src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -1060,6 +1061,78 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
         menu->addAction( newTableAction );
       }
     }
+
+    // SQL dialog
+    if ( std::unique_ptr<QgsAbstractDatabaseProviderConnection> conn( item->databaseConnection() ); conn )
+    {
+
+      if ( conn->capabilities().testFlag( QgsAbstractDatabaseProviderConnection::Capability::ExecuteSql ) )
+      {
+
+        QAction *sqlAction = new QAction( QObject::tr( "Run SQL command…" ), menu );
```

Comment:
Actually I opted for "command" instead of "query" because I assumed that query is for data retrieval only while "command" includes all DB operations like DROP, ALTER etc.

But I have no strong opinion.